src/lib.rs

```diff
@@ -98,52 +131,74 @@ impl TcpStream {
     /// If multiple address is given, the first successful socket is
     /// returned.
     pub fn connect<A: ToSocketAddrs>(addrs: A) -> Result<TcpStream> {
-        match addrs.to_socket_addrs()?.find_map(|addr| unsafe {
-            let fd = SocketHandle(ssvm_sock_open(
-                AddressFamily::from(addr) as u8,
+        match addrs.to_socket_addrs()?.find_map(|addrs| unsafe {
+            let mut fd: u32 = 0;
```

Comment:
`fd` has some special magic number: `0` -> `stdin`, `1` -> `stdout`, `2` -> `stderr`. Maybe we should use the default value from `3`?

Reply:
This is a mutable fd that its number will be given by POSIX connect() for the created new socket connection

Reply:
Got it.